### ISSUE TITLE
Calling "emsdk activate --global latest" hangs when setting environment variables.

### ISSUE THREAD

**Issue by timhutton** [emscripten-core/emsdk#138]
Platform: Windows 10, Python 2.7.13

Without the --global option the command runs very quickly and succeeds. With --global it hangs here:

```
...
Setting environment variables:
EMSDK = c:/emscripten

```

**WORKAROUND:** ```emsdk activate latest```

(From [this commit](https://github.com/juj/emsdk/commit/2ed7a605a90ca72f750ffd04df736e67d4523ff4) it looks like this acts globally anyway.)

**Comment by sbryfcz** [comment]
I have the same issue. Platform Windows 10. Python 3.6.3.

**Comment by juj** [comment]
With `--global` the Windows UAC elevation prompt should appear to ask for permission to permanently add the PATHs and environment variables to registry. If that does not happen and you're getting a hang before that line, I wonder if [running Windows Command Prompt in Administrator mode](https://www.howtogeek.com/194041/how-to-open-the-command-prompt-as-administrator-in-windows-8.1/) might help avoid the issue?

**Comment by sbryfcz** [comment]
The UAC prompt never appeared for me. And I tried in Adminstrator mode but nothing happened except it hanging.

**Comment by d07RiV** [comment]
Same issue. It worked fine on my other machine, but here no matter what I do, it hangs before setting LLVM_ROOT, and none of the variables it claims to have set appear in global environment. Tried both cmd and powershell in administrator mode.

**Comment by aplumly** [comment]
anyone find a fix for this?... apart from setting all the env variables by hand?

**Comment by roman-petrov** [comment]
Today I spent a hour trying to investigate this issue on my machine. As I see the hang caused by `win32api.SendMessage(win32con.HWND_BROADCAST, win32con.WM_SETTINGCHANGE, 0, 'Environment')`
call in `emsdk` Python script.

Found some on StackOverflow https://stackoverflow.com/questions/1951658/sendmessagehwnd-broadcast-hangs

Still not sure if there is a simple workaround, but maybe removing this system call should work. Or we can use 'SendMessageTimeout` a timeout in this call as they suggested on `StackOverflow`.

I am new with `emsdk`  so I am not sure what is the right solution, need help...
